lib/Mail/Milter/Authentication/Handler/SPF.pm

```diff
@@ -194,13 +198,153 @@ sub envfrom_callback {
         $self->log_error( 'SPF Error ' . $error );
         $self->{'failmode'} = 1;
         my $header = Mail::AuthenticationResults::Header::Entry->new()->set_key( 'spf' )->safe_set_value( 'temperror' );
-        $self->add_auth_header($header);
+        $self->{'spf_header'} = $header;
         $self->metric_count( 'spf_total', { 'result' => 'error' } );
     }
 }
 
+sub header_callback {
+    my ( $self, $header, $value ) = @_;
+
+    return unless exists $self->{'spfu_chain'};
+    return unless $self->{'dmarc_result'} eq 'pass';
```

Comment:
Go and look at this.  Is it possible to have a dmarc pass while in the header stage?